modules/core/src/main/scala/sangria/validation/QueryValidator.scala

```diff
@@ -63,13 +71,19 @@ class RuleBasedQueryValidator(
     rules: List[ValidationRule],
     errorsLimit: Option[Int]
 ) extends QueryValidator {
-  def validateQuery(schema: Schema[_, _], queryAst: ast.Document): Vector[Violation] = {
+  def validateQuery(schema: Schema[_, _], queryAst: ast.Document): Vector[Violation] =
+    validateQuery(schema, queryAst, errorsLimit)
+
+  def validateQuery(
+      schema: Schema[_, _],
+      queryAst: ast.Document,
+      errorsLimitOverride: Option[Int]): Vector[Violation] = {
     val ctx = new ValidationContext(
       schema,
       queryAst,
       queryAst.sourceMapper,
       new TypeInfo(schema),
-      errorsLimit)
+      errorsLimitOverride)
```

Comment:
It would take precedence over RuleBasedQueryValidator.errorsLimit so we could pass and use a single limit via `execute` method